Bento/Views/BentoReusableView.swift

```diff
@@ -1,10 +1,20 @@
+/// Existential type for reusable views (cell, header, footer) in `UITableView`, `UICollectionView`, etc.
 protocol BentoReusableView: AnyObject {
-    var containedView: UIView? { get set }
+    /// The main view to which you add your reusable view’s custom content.
+    /// - Note: This is normally system-provided as `self`'s subview.
     var contentView: UIView { get }
+
+    /// `component`-generated view that will be added to `contentView`.
```

Comment:
```suggestion
    /// The component root view that is currently installed in `contentView`. If the view is not compatible with the component to be bound, a new compatible view should be instantiated to replace it.
```

Reply:
Can you provide more info about `If the view is not compatible...` ?
I think the scenario example needs to be also documented.

Reply:
If the cell is previously bound with Component Apple, and now it is asked to be bound with Component Banana, the current `containedView` would not be compatible with that of Component Banana, and therefore a view needs to be instantiated for Component Banana.

Reply:
```suggestion
    /// The component root view that is currently added in `contentView`.
    /// If the existing view is no longer compatible with the newly set `component`,
    /// a new compatible view should be instantiated to replace it.
```

Slightly modified https://github.com/Babylonpartners/Bento/pull/142#discussion_r273426139 for simplicity.